datafusion/expr/src/udaf.rs

```diff
@@ -608,6 +608,60 @@ impl AggregateUDFImpl for AliasedAggregateUDFImpl {
         &self.aliases
     }
 
+    fn state_fields(&self, args: StateFieldsArgs) -> Result<Vec<Field>> {
```

Comment:
👍 nice - to avoid similar bugs in the future maybe we should also add a comment to `AggregateUDFImpl` that tries to remind people to add an implementation here. 

It would be cool if we could tell Rust not to create a default implementation for a particular `impl` so the compiler could do the check for us, but I don't know how to express that in Rust

Reply:
Added notes in https://github.com/apache/datafusion/pull/12098/commits/175b8d2b680af7e1e2c0563784946f55d6761a24. And yep - it would be nice to have a way of enforcing this, but I also couldn't figure out any way to do that :/